src/core/tranche/distributor/line.sol

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2019 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "./distributor.sol";
+
+contract LOC is Distributor {
+
+    constructor (address manager_, address flow_) Distributor(manager_, flow_) public {}
+
+    function balance() public auth line {
+        require(manager.poolClosing() == false);
+
+        int wad = manager.checkPile();
+        if (wad > 0) {
+            borrowFromTranches(uint(wad));
+        } else if (wad < 0) {
+            // takes extra money from Pile and repays tranche debt to avoid accumulation of fees
+            repayTranches(uint(wad*-1));
+        }
+    }
+
+    function borrowFromTranches(uint borrowAmount) private {
```

Comment:
I have an idea. I  think we can move this method to the Distributor base contract as well.

The two different distributors would only a different `balance` implementation. 

The LOC distributor would call borrow with the borrowAmount and the fixed distributor could call this method
with `borrowAmount = uint(-1)` which would result in borrowing all the available currency from all tranches.

What do you think?

Reply:
We could say the LOC is our default implementation of the distributor with the balance method. For the fixed one we use inheritance and overwrite the balance method. That's it.

We could remove the LOC contract.

Reply:
Even the balance method would be similar. For the repay we always use all the currency available in the pile.

Only for the borrow case the currencyAmount is different. In one distirbutor based on pile.Want in the other one `uint(-1)` to take it all.

We might could only have one Distibutor with a `takeItAll` flag for the two ways to get the currencyAmount